File: 3753-MaximumDifferenceBetweenEvenAndOddFrequencyI/3753-MaximumDifferenceBetweenEvenAndOddFrequencyI.py
# Last updated: 1/12/2026, 3:12:53 PM
class Solution:
    def maxDifference(self, s: str) -> int:
        arr = [0] * 26
        for c in s:
            arr[ord(c)-ord('a')] += 1
        maxOdd = max(c for c in arr if c%2==1)
        minEven = min(c for c in arr if c%2==0 and c > 0)
        return maxOdd - minEven

File: 3753-MaximumDifferenceBetweenEvenAndOddFrequencyI/3753-MaximumDifferenceBetweenEvenAndOddFrequencyI.cpp
// Last updated: 1/12/2026, 3:12:51 PM
class Solution {
public:
    int maxDifference(string s) {
        vector<int> cnt(26, 0);
        for(char c : s) {
            cnt[c - 'a']++;
        }
        int maxOdd = INT_MIN, minEven = INT_MAX;
        for(int i = 0; i < 26; i++) {
            if(cnt[i] == 0) continue;
            if(cnt[i] & 1) {
                maxOdd = max(maxOdd, cnt[i]);
            } else {
                minEven = min(minEven, cnt[i]);
            }
        }
        return maxOdd - minEven;
    }
};